src/dvclive/dvc.py

```diff
@@ -177,3 +181,29 @@ def get_dvc_stage_template(live):
     dvcyaml = output.getvalue()
     output.close()
     return dvcyaml
+
+
+def update_models(path, attrs, models):
+    from dvc.repo.artifacts import name_is_compatible
+
+    if attrs["name"] is None:
+        name = find_unused_name(path, models)
+    else:
+        name = attrs.pop("name")
+
+    attrs["path"] = path
+    if name_is_compatible(name):
+        if name not in models:
+            models[name] = {}
+        for k, v in attrs.items():
+            if v is not None:
+                models[name][k] = v
+    else:
+        logger.warning(
+            "Can't use '%s' as artifact name (ID)."
+            " You can use letters and numbers, and use '-' as separator"
+            " (but not at the start or end)."
+            " The first character must be a letter.",
+            name,
```

Comment:
Yeah, this looks bad unless we can condense the logic more 😓 . @aguschin WDYT? Can we explain it better? Do we really need it to be this convoluted?